### PR TITLE
Use "hello" command where possible

### PR DESCRIPTION
**Pull request by jyemin**
There are two scenarios where "hello" command can be used:

1. When the application declares an API version
3. When the response to "ismaster" contains helloOk: true

Additionally, use OP_MSG when the application declares an API version,
even when maxWireVersion is unknown during the handshake

JAVA-4092
JAVA-4157

### CODE REVIEW

File: driver-core/src/test/unit/com/mongodb/internal/connection/MessageHelper.java [hunk 1]
```diff
@@ -105,6 +114,10 @@ public static String getApiVersionField() {
         return getServerApi() == null ? "" : ", \"apiVersion\": \"" + getServerApi().getVersion().getValue() + "\"";
     }
 
+    public static String getDbField(final String databaseName) {
```
Comment (rozza) [inline driver-core/src/test/unit/com/mongodb/internal/connection/MessageHelper.java:117]: Is this related to hello?

**Review by rozza — COMMENTED** [review]
One question about the db change in auth.

File: driver-core/src/test/unit/com/mongodb/internal/connection/MessageHelper.java (hunk 1 above)
Comment (jyemin) [reply driver-core/src/test/unit/com/mongodb/internal/connection/MessageHelper.java:117]: Loosely.  It's related to the change in CommandMessage.getOpCode where we now use OP_MSG if API version is set.  If API version is non-null, we use OP_MSG, and when we use OP_MSG the command document includes a $db field.

**Review by rozza — APPROVED** [review]
LGTM

File: driver-core/src/test/unit/com/mongodb/internal/connection/MessageHelper.java (hunk 1 above)
Comment (rozza) [reply driver-core/src/test/unit/com/mongodb/internal/connection/MessageHelper.java:117]: ACK - thanks for the clarification
